presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -176,7 +178,7 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
         return schemaName.map(Collections::singletonList)
                 .orElseGet(metastore::getAllDatabases)
                 .stream()
-                .flatMap(schema -> metastore.getAllTables(schema).stream()
+                .flatMap(schema -> metastore.getTablesMatchingParameter(schema, TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE).stream()
```

Comment:
Is the match case-sensitive or case-insensitive?

On our side we do case-insensitive, `file` metastore it's case-sensitive and HMS thrift call may be case-sensitive (or it may actually depend on hms backend db collation).

Reply:
@electrum thoughts?

Reply:
@findepi @linxingyuan1102 The change introduces the problematic behavior as AFAIK case-sensitivity depends on HMS backend db collation and for derby db it is case-sensitive. This leads to iceberg tables that are created in spark using an upper case 'ICEBERG' type become hidden in `show tables` while still valid and usable in queries.

Reply:
@vrozov thanks for your comment. As you see, this was assumed as probable.
May I suggest to move discussion to  `#iceberg` [slack](https://prestosql.io/slack.html) channel?
Otherwise a conversation under a closed PR may escape attention of some people.

Reply:
@findepi Both #iceberg slack channel or new issue #1592 is fine with me. I commented here, to make it easier to refer to the root cause of the problem.